### PR TITLE
Documentation: Add a getting started contributing guide

### PR DESCRIPTION
**Pull request by circl-lastname**
This document aims to be a begginer-friendly summary of the existing contribution requirements for Ladybird, intended to be linked to new-comers.

Relies on LadybirdBrowser/ladybird.org#71

### CODE REVIEW

File: Documentation/GettingStartedContributing.md
Comment (AtkinsSJ) [inline Documentation/GettingStartedContributing.md:4]: s/begginer/beginner/

File: Documentation/GettingStartedContributing.md
Comment (sideshowbarker) [inline Documentation/GettingStartedContributing.md:4]:
```suggestion
This document aims to be a beginner-friendly summary of the existing contribution requirements for Ladybird; remember to read the linked documentation for more information.
```

File: Documentation/GettingStartedContributing.md
Comment (sideshowbarker) [inline Documentation/GettingStartedContributing.md:9]:
```suggestion
It is recommend you read the FAQ in case it already answers any questions you have; there is also a [Discord server](https://discord.gg/nvfjVJ4Svh), which is the preferred way to ask the maintainers and community for help.
```

File: Documentation/GettingStartedContributing.md
Comment (sideshowbarker) [inline Documentation/GettingStartedContributing.md:17]:
```suggestion
Ladybird must be built from source during this pre-alpha stage of development, and currently only supports Linux and macOS natively; running it on Windows requires WSL. Please carefully follow the steps outlined in the build instructions.
```

File: Documentation/GettingStartedContributing.md
Comment (sideshowbarker) [inline Documentation/GettingStartedContributing.md:28]:
```suggestion
* **By finding failing WPT tests** on [WPT.fyi](https://staging.wpt.fyi/results/?label=master&product=ladybird). Note that you do not need to submit separate issues for these, but fixes are welcome
```

File: Documentation/GettingStartedContributing.md
Comment (sideshowbarker) [inline Documentation/GettingStartedContributing.md:44]:
```suggestion
* Make correctly formatted, atomic commits (building the project at every commit should succeed).
```

File: Documentation/GettingStartedContributing.md
Comment (sideshowbarker) [inline Documentation/GettingStartedContributing.md:45]:
```suggestion
* Use the `rebase` function of `git` to keep your local branch up-to-date with master, and to edit existing commits.
```

File: Documentation/GettingStartedContributing.md
Comment (sideshowbarker) [inline Documentation/GettingStartedContributing.md:46]:
```suggestion
* Use the AK library instead of the C++ STL.
```

File: Documentation/GettingStartedContributing.md
Comment (sideshowbarker) [inline Documentation/GettingStartedContributing.md:47]:
```suggestion
* Discuss and resolve any reviews from the maintainers.
```

File: Documentation/GettingStartedContributing.md
Comment (ADKaster) [inline Documentation/GettingStartedContributing.md:50]: We should promote the full contributing.md more as the "but actually read and follow this before submitting your PR", rather than suggesting it as a 'recommended follow-up', as this section of links appears to be at first glance.

File: Documentation/GettingStartedContributing.md
Comment (ADKaster) [reply Documentation/GettingStartedContributing.md:47]: Reviews from 'pull request reviewers' ? We don't exactly want to set an expectation that only maintainers can leave actionable PR reviews. Contributors should be encouraged to review code if they so fancy. As well as external reviews from subject matter experts (someone from WhatWG/W3C/TC39, a crypto expert, etc.)

**Review by ADKaster — COMMENTED** [review]
Overall I think this is a good idea. We've been getting a lot of new-to-github and new-to-c++ questions on Discord lately. A future update to this document could go over how to setup one's fork and local branches for successful long-term contributions. Every time I see a pull request from someone's master/main branch I cry a little :'(

**Comment by circl-lastname** [comment]
There are a few thing I don't like about it in the current state:
* Like Andrew pointed out, a contributor might read the Cherry-picked summary of the guidelines, and ignore the full ones available
* I've been avoiding using the pronoun "We" to refer to the Ladybird project, for instance "It is recommended" rather than "We recommend". Some things may be phrased awkwardly as a result
* This still doesn't explicitly answer "I don't know C++, can I contribute?" and "How do I find beginner friendly issues?" 

I will probably
* Change the structure to explicity mention reading the full documentation with the link closer by
* Add more guidance to the issues section
* Split up the code submission section into more parts, directly addressing the level of C++ knowledge needed, and other ways the reader can help

**Review by sideshowbarker — COMMENTED** [review]
:thumb: Non-maintainer LGTM

File: Documentation/SmartPointers.md
Comment (jamierocks) [inline Documentation/SmartPointers.md:1]: I think qualifying this as 'Ladybird smart pointers' or 'AK smart pointers' is a good idea, just to immediately clarify that we don't use smart pointers from the C++ standard library.

**Comment by circl-lastname** [comment]
Made some final changes to the docs it links to, hopefully this is good enough for a marge.

Now the only thing blocking is my website PR for the FAQ link, and later I will add a command to the Discord bot

**Comment by ADKaster** [comment]
This seems reasonable now. We just got our changes merged from staging.wpt.fyi up to prod wpt.fyi, but we can fix that up after merge.

**Review by ADKaster — APPROVED** [review]

**Pull request merged**
